### PR TITLE
Add/Edit Container Groups

### PR DESCRIPTION
**Pull request by nixocio**
Add/Edit container groups.

See: https://github.com/ansible/awx/issues/7955

image

### CODE REVIEW

File: awx/ui_next/src/screens/InstanceGroup/shared/ContainerGroupForm.jsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        tooltip={i18n._(
+          t`Credential to authenticate with Kubernetes or OpenShift.  Must be of type "Kubernetes/OpenShift API Bearer Token”.`
+        )}
+      />
+
+      <FormGroup
+        fieldId="container-groups-option-checkbox"
+        label={i18n._(t`Options`)}
+      >
+        <FormCheckboxLayout>
```
Comment (jakemcdermott) [inline awx/ui_next/src/screens/InstanceGroup/shared/ContainerGroupForm.jsx:60]: @nixocio 

In terms of functionality, this is very similar to the notification config overrides we do on the notification form and the UX should be consistent for both features.

- In the 3.8 ui we use a toggle-switch for container groups and notifications
- The mockups [show a checkbox](https://tower-mockups.testing.ansible.com/patternfly/instance-groups/instance-groups-container-edit/) for container groups but don't specify any UX for notifications
- [We intend to use a switch](https://github.com/ansible/awx/pull/7967/files#) for notifications in ui_next


Personally, I prefer the toggle switch but I think the important thing is that we're consistent with the type of input field we use for configuration overrides.

cc @trahman73 @mabashian @keithjgrant  thoughts?

**Review by jakemcdermott — CHANGES_REQUESTED** [review]
The code, functionality, and tests look good to me. Nice work!

I would like for https://github.com/ansible/awx/pull/8070#discussion_r483779879 to be resolved (in some fashion) prior to merging. Depending on what is decided, this may mean changing the override field to a toggle.

File: awx/ui_next/src/screens/InstanceGroup/shared/ContainerGroupForm.jsx (hunk 1 above)
Comment (nixocio) [reply awx/ui_next/src/screens/InstanceGroup/shared/ContainerGroupForm.jsx:60]: There was a slack discussion about the need to use `checkbox`. I think the mock up was updated upon the outcome of that discussion.

Comment (jakemcdermott) [reply awx/ui_next/src/screens/InstanceGroup/shared/ContainerGroupForm.jsx:60]: :+1: Nice. Sounds good to me.

**Review by jakemcdermott — APPROVED** [review]

**Review by jakemcdermott — APPROVED** [review]
https://github.com/ansible/awx/pull/8070#discussion_r483803623

**Comment by dsesami** [comment]
I'm sitting with @nixocio on Monday 9/14 to discuss this PR in detail.

**Review by dsesami — APPROVED** [review]
Covered this at length with @nixocio . Looks great. :+1: Will follow up on some improvement ideas at dev sync.

**Pull request merged**